src/components/pages/contact/contact-form.tsx

```diff
@@ -1,148 +1,165 @@
 import React, { useState } from 'react';
+import { TextInput } from '../../forms/text-field/text-input';
+import { TextArea } from '../../forms/text-field/text-area';
+import { useTranslation } from 'gatsby-plugin-react-i18next';
 import { useForm } from 'react-hook-form';
-import {
-  ButtonText,
-  ErrorMessageSend,
-  InputField,
-  SendButton,
-  SentButton,
-  Sup,
-} from '../../legacy/form/controls';
-import { Grid, GridItem } from '../../legacy/grid/grid';
-import { Link } from '../../legacy/links/links';
-import { CaptionText } from '../../legacy/typography';
-import { Trans, useTranslation } from 'gatsby-plugin-react-i18next';
+import { Button } from '../../ui/buttons/button';
+import { SIMPLE_EMAIL_PATTERN } from '../../forms/constants';
 import styled from 'styled-components';
-import { SIMPLE_EMAIL_PATTERN } from '../../legacy/form/constants';
-import { CheckmarkIcon } from '../../legacy/icons/form-icons/checkmark';
-import { RightArrowIcon } from '../../legacy/icons/form-icons/right-arrow';
+import { TextStyles } from '../../typography';
+import { theme } from '../../layout/theme';
+import { Link } from '../../legacy/links/links';
+import { up } from '../../support/breakpoint';
 import { HoneypotField } from './honeypot';
 
-type RequestStatus = 'pending' | 'submitting' | 'success' | 'error';
 const API_ENDPOINT = '/api/contact-form';
 
-const StyledCaptionText = styled(CaptionText)`
-  color: inherit;
+const Grid = styled.div`
+  margin-bottom: 24px;
+  gap: 24px;
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
```

Comment:
I see your shortcut here but please don't use a 12 column based grid for that simple case. I have imagined a markup like that.


```
<FormLayout>
  <Name/>
  <Email/>
  <Message/>
</FormLayout>

<HoneyPot/>
<MandatoryNotes/>
<Submit/>

<Errors/>

```

where `FormLayout` defines the two column grid and `Message` stretch itself across the two columns with grid-column set to -1/1. This removes a lot of noise in the markup.

I love to plan and design my code with this kind of API or markup driven approach. I can ask myself what would I love to interact with in the future. I write down a neat structure and see if I can map the reality to it. Maybe this helps you here to refactor the current markup which doesn't look very friendly to any future developer touching the contact form.

Reply:
Greta new structure and I'm happy that you picked up my recommendations so it made sense I guess 🙏